service-registry/src/main/java/io/servicecomb/serviceregistry/task/MicroserviceRegisterTask.java

```diff
@@ -143,12 +146,53 @@ private boolean registerSchemas() {
       boolean exists = srClient.isSchemaExist(microservice.getServiceId(), schemaId);
       LOGGER.info("schemaId {} exists {}", schemaId, exists);
       if (!exists) {
-        if (!srClient.registerSchema(microservice.getServiceId(), schemaId, content)) {
-          return false;
+        if (serviceCenterEnvIsDev()) {
+            if (!srClient.registerSchema(microservice.getServiceId(), schemaId, content)) {
```

Comment:
product mode, will not register schema?

Reply:
done, if not exist, just register schema